### PR TITLE
[jenkins-jobs] Add MongoDB 4.2, remove alpine tests

### PR DESCRIPTION
**Pull request by jpechane**

### CODE REVIEW

File: jenkins-jobs/job-dsl/postgresql_matrix_test.groovy [hunk 1]
```diff
@@ -12,7 +12,7 @@ matrixJob('debezium-postgresql-matrix-test') {
          ''')
 
     axes {
-        text('POSTGRES_VERSION', '10', '11', '9.6-alpine', '10-alpine', '11-alpine')
+        text('POSTGRES_VERSION', '10', '11')
```
Comment (gunnarmorling) [inline jenkins-jobs/job-dsl/postgresql_matrix_test.groovy]: Didn't we add 12 recently? And why is Alpine removed?

Comment (jpechane) [reply jenkins-jobs/job-dsl/postgresql_matrix_test.groovy]: @gunnarmorling Good point with 12, the changes were done in parallel so it was lost.
Regarding alpine - we've remove postgis support from them. So we need to refactor postgres tests to extract postgis tests so they can be filtered.

Comment (gunnarmorling) [reply jenkins-jobs/job-dsl/postgresql_matrix_test.groovy]: > So we need to refactor postgres tests to extract postgis tests so they can be filtered.

Either that; or would it be fine to simply run the Tests against the full images? I'm torn.

Comment (jpechane) [reply jenkins-jobs/job-dsl/postgresql_matrix_test.groovy]: @gunnarmorling We have no full apline images and if you recall we are unable to provide them as alpine requires increaseing number of dependencis of postgis to be built from the source. That was the reason in the first case to remove postgis from alpine

Comment (gunnarmorling) [reply jenkins-jobs/job-dsl/postgresql_matrix_test.groovy]: Understood. What I meant is, do we actually need to do this one:

>  So we need to refactor postgres tests to extract postgis tests so they can be filtered.

(as a follow-up). Or are we fine with this PR as is, i.e. run the test suite against the other images but not Alpine?

Comment (jpechane) [reply jenkins-jobs/job-dsl/postgresql_matrix_test.groovy]: I'd prefer to do the follow-up - when time permits

**Comment by gunnarmorling** [comment]
Applied. Also logged https://issues.jboss.org/browse/DBZ-1641 for making the PostGIS tests skippable. Thanks, @jpechane!

**Pull request merged**